mypy/subtypes.py

```diff
@@ -1376,8 +1376,14 @@ def visit_literal_type(self, left: LiteralType) -> bool:
             return self._is_proper_subtype(left.fallback, self.right)
 
     def visit_overloaded(self, left: Overloaded) -> bool:
-        # TODO: What's the right thing to do here?
-        return False
+        if isinstance(self.right, Overloaded):
+            for right_item in self.right.items:
+                if not any(self._is_proper_subtype(left_item, right_item)
```

Comment:
In general, order of overloads matters (think multiple inheritance, or an overload that shadows another one). I think I'd just check that the overloads match exactly, allowing for more items at the end of `right`. Happily, this would also mean that the check would no longer be quadratic.

Reply:
I do not understand why the order of overloads should matter, or why they should overlap exactly, overloads do not have order in other languages I'm aware of. If there was a decision made in favor of the order of overloads having significance, I think it would restrict otherwise valid structural typing, for example this case https://mypy-play.net/?mypy=latest&python=3.10&flags=strict&gist=1a5795445d9ce093cd9878f79e45b48d , it is valid under this implementation and I do not think it would be if I re-implemented it the way you have described.

I would think that ideally, although perhaps naively, a method/function would resolve to the most specific match instead of the first match in the order of overloads.